pydantic/generics.py

```diff
@@ -27,7 +27,7 @@ def __class_getitem__(  # type: ignore
             raise TypeError('Cannot parameterize a concrete instantiation of a generic model')
         if not isinstance(params, tuple):
             params = (params,)
-        if any(isinstance(param, TypeVar) for param in params):  # type: ignore
+        if cls == GenericModel and any(isinstance(param, TypeVar) for param in params):  # type: ignore
```

Comment:
I think `if cls is GenericModel and ...` might be preferable here.

Reply:
Fixed in 0b1bb6df58304e9ba7777bc77545e265bd0ba870